File: python/secret-handshake/secret_handshake.py
ACTIONS = [
    'wink',
    'double blink',
    'close your eyes',
    'jump',
]

def handshake(code):
    actions = [action for index, action in enumerate(ACTIONS) if code & 2**index]
    if code & 2**len(ACTIONS):
        actions.reverse()
    return actions

def secret_code(actions):
    codes = [2**ACTIONS.index(action) for index, action in enumerate(actions) if action in ACTIONS]
    if codes != sorted(codes):
        codes += [2**len(ACTIONS)]
    return sum(codes)

File: javascript/secret-handshake/secret-handshake.js
const ACTIONS = [
    'wink',
    'double blink',
    'close your eyes',
    'jump',
];

export const secretHandshake = (code) => {
    if (typeof code !== 'number')
        throw new TypeError('Handshake must be a number');

    const actions = ACTIONS.filter((_, i) => code & 1 << i);

    return code & 1 << ACTIONS.length
        ? actions.reverse()
        : actions;
};
